lib/cli.js

```diff
@@ -21,82 +21,70 @@ var fs = require('fs');
 var getGitFolder = require('./getGitFolder');
 var validateMessage = require('../index');
 
-// hacky start if not run by mocha :-D
-// istanbul ignore next
-if (process.argv.join('').indexOf('mocha') === -1) {
-  var bufferToString = function (buffer) {
-    var hasToString = buffer && typeof buffer.toString === 'function';
-    return hasToString && buffer.toString();
-  };
+var commitMsgFileOrText = process.argv[2];
+var commitErrorLogPath = process.argv[3];
 
-  var getFileContent = function (filePath) {
-    try {
-      var buffer = fs.readFileSync(filePath);
-      return bufferToString(buffer);
-    } catch (err) {
-      // Ignore these error types because is most likely it is validating
-      // a commit from a text instead of a file
-      if(err && err.code !== 'ENOENT' && err.code !== 'ENAMETOOLONG') {
-        throw err;
-      }
-    }
-  };
+// On running the validation over a text instead of git files such as COMMIT_EDITMSG and GITGUI_EDITMSG
+// is possible to be doing that the from anywhere. Therefore the git directory might not be available.
+var gitDirectory;
+try {
+  gitDirectory = getGitFolder();
+
+  if (!commitErrorLogPath) {
+    commitErrorLogPath = gitDirectory + '/logs/incorrect-commit-msgs';
+  }
+} catch (err) {}
+
+validate(commitMsgFileOrText);
+
+
+function bufferToString(buffer) {
+  var hasToString = buffer && typeof buffer.toString === 'function';
+
+  return hasToString && buffer.toString();
+}
+
+function getCommit(msgFileOrText) {
+  return getContentFromFile(msgFileOrText)
+    || !msgFileOrText && getContentFromFile('COMMIT_EDITMSG')
+    || { message: msgFileOrText };
```

Comment:
This should be functionally equivalent. But I'd also not use the logical not operator. Instead hard do the checks to `null` as that is returned if not found. Minuscule optimization to save the engine some time. But also providing more legible code in the future since you know exactly what the check is looking for all the time. 

```javascript
var commit = getContentFromFile(msgFileOrText);

if (commit === null) {
  commit = getContentFromFile('COMMIT_EDITMSG');
}

if (commit === null) {
  commit = { message: msgFileOrText };
}

return commit;
```